src/Controller/CategoriesController.php

```diff
@@ -45,63 +42,51 @@ public static function create(ContainerInterface $container) {
   }
 
   /**
-   * Categories list with filters by content type/bundle.
-   *
-   * Provides a list of categories uuid's that contains
-   * videos for specific bundle.
-   *
-   * @param \Symfony\Component\HttpFoundation\Request $request
-   *   The request.
+   * Categories tree.
    *
    * @return \Symfony\Component\HttpFoundation\JsonResponse
-   *   Json Array with taxonomy uuid's.
+   *   Json Object with taxonomy tree data.
    */
-  public function list(Request $request): JsonResponse {
-    $request_data = UrlHelper::parse($request->getUri());
-    if (empty($request_data['query']) || !isset($request_data['query']['type'])) {
-      return new JsonResponse(['message' => 'Missed "type" query param.'], 400);
-    }
-
-    if ($request_data['query']['type'] !== 'all' && !isset($request_data['query']['bundle'])) {
-      return new JsonResponse(['message' => 'For specified "type" should be "bundle" query param.'], 400);
-    }
+  public function list(): JsonResponse {
+    $bundles_entity_types = [
+      'gc_video' => 'node',
+      'live_stream' => 'eventinstance',
+      'virtual_meeting' => 'eventinstance',
+      'vy_blog_post' => 'node',
+    ];
 
     $result = [];
-    switch ($request_data['query']['type']) {
-      case 'all':
-        $data1 = $this->getNodeCategories(['gc_video', 'vy_blog_post']);
-        $data2 = $this->getEventCategories(['live_stream', 'virtual_meeting']);
-        $result = array_unique(array_merge($data1, $data2));
-        break;
-
-      case 'node':
-        $result = $this->getNodeCategories([$request_data['query']['bundle']]);
-        break;
-
-      case 'eventinstance':
-        $result = $this->getEventCategories([$request_data['query']['bundle']]);
-        break;
+    foreach ($bundles_entity_types as $bundle => $type) {
+      switch ($type) {
+        case 'node':
+          $result[$bundle] = $this->getNodeCategories($bundle);
+          break;
+
+        case 'eventinstance':
+          $result[$bundle] = $this->getEventCategories($bundle);
+          break;
+      }
     }
 
-    return new JsonResponse(array_values($result));
+    return new JsonResponse($this->buildCategoriesTree($result));
   }
 
   /**
-   * Categories list filtered by node content and bundle.
+   * Categories list filtered by node's bundle.
    *
-   * @param array $bundles
-   *   Node types for filtering results.
+   * @param string $bundle
+   *   Node type for filtering results.
    *
    * @return array
-   *   Array with taxonomy uuid's.
+   *   Array with taxonomy id's.
    */
-  protected function getNodeCategories(array $bundles): array {
+  protected function getNodeCategories(string $bundle): array {
```

Comment:
Why you changed this to a single value?

Reply:
The current logic is: to get the categories list for each bundle and then combine this information when building the categories tree, and then use the bundles' info on the Vue app side to filter categories by the content existing